app/javascript/components/mentoring/discussion/MarkAsNothingToDoButton.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect } from 'react'
+import { useMutation } from 'react-query'
+import { Loading } from '../../common'
+import {
+  ErrorBoundary,
+  useErrorHandler,
+  FallbackProps,
+} from 'react-error-boundary'
+import { APIError } from '../../types'
+
+type ComponentProps = {
+  endpoint: string
+}
+
+const ERROR_MESSAGE_TIMEOUT = 500
+
+export const MarkAsNothingToDoButton = (props: ComponentProps): JSX.Element => {
+  return (
+    <ErrorBoundary FallbackComponent={ErrorFallback}>
+      <Component {...props} />
+    </ErrorBoundary>
+  )
+}
+
+const ErrorFallback = ({ error, resetErrorBoundary }: FallbackProps) => {
+  useEffect(() => {
+    setTimeout(resetErrorBoundary, ERROR_MESSAGE_TIMEOUT)
+  }, [resetErrorBoundary])
+
+  return (
+    <div>
+      <p>{error.message}</p>
+    </div>
+  )
+}
+
+const Component = ({ endpoint }: ComponentProps): JSX.Element | null => {
+  const handleError = useErrorHandler()
+  const [mutation, { status, error }] = useMutation(() => {
```

Comment:
@ErikSchierboom I just found out that `react-query` has this `useMutation` hook that allows us to send requests that aren't idempotent. We currently have a own version of this called `sendRequest`, but I'd like to use `useMutation` moving forward. I haven't zeroed in on a good pattern, but as I use it, I hope to eventually get it. I also plan to refactor our other components to use this, once we get the time to do so.

Reply:
I don't think you meant to use the word `idempotent` here, but instead "are not GET or HEAD" request.

Reply:
That's a great find!